app/admin/jobs/routine/[id]/page.tsx

```diff
@@ -0,0 +1,164 @@
+"use client";
+
+import React, { useEffect, useState } from "react";
+import { getVellumSandboxUrl } from "@/app/lib/vellum-utils";
+
+interface RoutineJob {
+  id: string;
+  name: string;
+  cronExpression: string;
+  enabled: boolean;
+  createdAt: string;
+}
+
+export default function RoutineJobDetailPage({
+  params,
+}: {
+  params: Promise<{ id: string }>;
+}) {
+  const [routineJob, setRoutineJob] = useState<RoutineJob | null>(null);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+  const [testing, setTesting] = useState(false);
+  const [testResult, setTestResult] = useState<{ success: boolean; outputs?: unknown; message: string } | null>(null);
+  const [resolvedParams, setResolvedParams] = useState<{ id: string } | null>(null);
+
+  useEffect(() => {
+    params.then(setResolvedParams);
+  }, [params]);
+
+  useEffect(() => {
+    if (!resolvedParams) return;
+
+    const fetchRoutineJob = async () => {
+      try {
+        const response = await fetch(`/api/jobs/routine/${resolvedParams.id}`);
+        if (!response.ok) {
```

Comment:
We should be able to use server actions here instead of an api route in a useeffect